### PR TITLE
Added improved active filters, also made the filter functions globally available

### PR DESCRIPTION
**Pull request by BobWez98**
This adds a new frontend for the active filtering. Sadly we need a Vue component for this to map out the active filtering as there's some irregularity's in there like our Category filter, price and color filter.

I also made the Vue filters globally available so we can use them outside of templates, like the `url()` method already was.

This PR uses the colors added in #580

### CODE REVIEW

File: src/Http/Controllers/CategoryController.php [hunk 1]
```diff
@@ -10,6 +10,7 @@ public function show(int $categoryId)
         $category = $categoryModel::findOrFail($categoryId);
 
         config(['frontend.category' => $category->only('entity_id')]);
+        config(['frontend.subcategories' => $category->subcategories->mapWithKeys(fn ($item, $key) => [$item->entity_id => $item->name])]);
```
Comment (royduin) [inline src/Http/Controllers/CategoryController.php:13]: I think pluck does the same: `->pluck('name', 'entity_id')`

File: resources/views/listing/partials/filter/selected.blade.php [hunk 2]
```diff
@@ -1,20 +1,32 @@
-<selected-filters :inner-class="{ button: '!hidden last:!inline-flex' }">
-    <x-rapidez::button
-        class="w-full mb-3 text-sm"
-        slot-scope="{ clearValues, selectedValues, components }"
-        v-on:click="clearValues"
+<selected-filters :inner-class="{ button: '!inline-flex' }">
```
Comment (royduin) [inline resources/views/listing/partials/filter/selected.blade.php:1]: Is that prop / class still used now we're overwriting the slot?

File: config/rapidez/frontend.php [hunk 3]
```diff
@@ -21,6 +22,9 @@
         // 'material',
     ],
 
+    // The attribute of the color swatch, used in the active filtering
+    'color_attribute' => 'color',
```
Comment (royduin) [inline config/rapidez/frontend.php:26]: How does this work? From Magento you can have multiple swatch attributes with their own (color) values.

**Review by royduin — CHANGES_REQUESTED** [review]

File: resources/js/components/Listing/Filters/CustomSelected.vue [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                    ) {
+                        result = this.selectedValues[filterKey].value.map((selected) => selected)
+                    }
+
+                    if (filterKey === 'price') {
+                        result = price(this.selectedValues[filterKey].value[0]) + ' - ' + price(this.selectedValues[filterKey].value[1])
+                    }
+
+                    
+                    if (Object.keys(swatches.value).includes(filterKey)) {
```
Comment (BobWez98) [inline resources/js/components/Listing/Filters/CustomSelected.vue:41]: I removed the Color config @royduin . Every swatch can be used as a "active" filter now. Even though color filters are circles with the color as background. I think the label will suffice as "active filter" ( E.G. "Blue") as most webshops do this.

File: resources/views/listing/partials/filter/selected.blade.php (hunk 2 above)
Comment (BobWez98) [reply resources/views/listing/partials/filter/selected.blade.php]: nope

File: config/rapidez/frontend.php (hunk 3 above)
Comment (BobWez98) [reply config/rapidez/frontend.php]: See my comment here; https://github.com/rapidez/core/pull/587/files#diff-e28427a88f3021679abe0aa8a08b6796959c85a443f2b127192669829c0a663aR41

File: resources/js/components/Listing/Filters/CustomSelected.vue [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+                            ? this.selectedValues[filterKey].value.length
+                            : !!this.selectedValues[filterKey].value)
+                    )
+                })
+                .flatMap((filterKey) => {
+                    let result
+
+                    if (
+                        Array.isArray(this.selectedValues[filterKey].value) &&
+                        !['category', 'price', ...Object.keys(swatches.value)].includes(filterKey)
```
Comment (royduin) [inline resources/js/components/Listing/Filters/CustomSelected.vue:32]: Can't we do something with early returns or a switch case so we don't have to specify this list here?

File: resources/js/vue-components.js [hunk 6]
```diff
@@ -28,3 +28,4 @@ Vue.component('listing', () => import('./components/Listing/Listing.vue'))
 Vue.component('checkout', () => import('./components/Checkout/Checkout.vue'))
 Vue.component('checkout-success', () => import('./components/Checkout/CheckoutSuccess.vue'))
 Vue.component('popup', () => import('./components/Popup.vue'))
+Vue.component('custom-selected', () => import('./components/Listing/Filters/CustomSelected.vue'))
```
Comment (royduin) [inline resources/js/vue-components.js:31]: Maybe rename it to something better, selected-filters-values?

File: resources/js/components/Listing/Filters/SelectedFiltersValues.vue [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+        },
+    },
+
+    render() {
+        return this.$scopedSlots.default(this)
+    },
+
+    computed: {
+        activeFilters() {
+            return Object.keys(this.selectedValues)
```
Comment (BobWez98) [inline resources/js/components/Listing/Filters/SelectedFiltersValues.vue:18]: Tried to make it as short as possible now @royduin

**Comment by royduin** [comment]
@indykoning, what do you think? 😇

File: resources/js/components/Listing/Filters/SelectedFiltersValues.vue [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+                    return showFilter && (Array.isArray(value) ? value.length : !!value)
+                })
+                .reduce((result, filterKey) => {
+                    let { value } = this.selectedValues[filterKey]
+
+                    if (filterKey === 'category') {
+                        return result.concat({ code: filterKey, value: config.subcategories[value] })
+                    }
+
+                    if (filterKey === 'price') {
```
Comment (indykoning) [inline resources/js/components/Listing/Filters/SelectedFiltersValues.vue:30]: Is this only possible with a price filterkey or is this something you can choose yourself, wether you want it to be a min/max filter?

**Review by indykoning — APPROVED** [review]
LGTM, just one question 🙂

**Comment by BobWez98** [comment]
Currently Rapidez does not support other slider type filters than price, see; https://github.com/rapidez/core/blob/master/resources/views/listing/partials/filter/price.blade.php . I'll make an internal note for this to be developed.

**Comment by royduin** [comment]
Do we need this in v2 or is just v3 fine?

**Comment by BobWez98** [comment]
I think v3 is fine.

**Pull request merged**